libsupport/include/katana/JSONTracer.h

```diff
@@ -0,0 +1,82 @@
+#ifndef KATANA_LIBSUPPORT_KATANA_JSONTRACER_H_
+#define KATANA_LIBSUPPORT_KATANA_JSONTRACER_H_
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <variant>
+#include <vector>
+
+#include "katana/ProgressTracer.h"
+#include "katana/config.h"
+
+namespace katana {
+
+class KATANA_EXPORT JSONTracer : public ProgressTracer {
+  JSONTracer(uint32_t host_id = 0, uint32_t num_hosts = 1)
```

Comment:
nit: as a matter of style, private methods go after public ones, e.g.,

```cpp
class Foo {
public:

private:
}
```

https://google.github.io/styleguide/cppguide.html#Declaration_Order